### PR TITLE
[MINI-4882] Interface for MAUID v2

### PR DESCRIPTION
**Pull request by laubengaier**
# Description
Implementation for MAUID v2
- adding `getContactId` and `getMauid`

*Waiting for JS SDK Release*

## Links
https://jira.rakuten-it.com/jira/browse/MINI-4882

# Checklist
- [x] I have read the [contributing guidelines](CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [ ] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [ ] I ran `fastlane ci` without errors

### CODE REVIEW

File: Sources/Classes/core/MiniAppAnalytics.swift [hunk 1]
```diff
@@ -12,6 +12,8 @@ internal enum MiniAppRATEvent: String, CaseIterable {
     case close
     case signatureFailure = "mini_app_signature_validation_fail"
     case getUniqueId = "mini_app_get_unique_id"
+    case getContactId = "mini_app_get_contact_id"
+    case getMauid = "mini_app_get_mauid_id"
```
Comment (munir-rakuten) [inline Sources/Classes/core/MiniAppAnalytics.swift:16]: can we rename `mini_app_get_mauid_id` to `mini_app_get_mauid`? I use `mini_app_get_mauid` in android.

Comment (munir-rakuten) [inline Sources/Classes/core/MiniAppAnalytics.swift:16]: can we rename `mini_app_get_mauid_id` to `mini_app_get_mauid`? I use `mini_app_get_mauid` in android.

Comment (laubengaier) [reply Sources/Classes/core/MiniAppAnalytics.swift]: yeah sure 👍  fits better

File: Sources/Classes/core/JavascriptBridge/MiniAppJavaScriptEnumerations.swift [hunk 2]
```diff
@@ -2,6 +2,8 @@ import UIKit
 
 enum MiniAppJSActionCommand: String {
     case getUniqueId
+    case getContactId
+    case getMauid
```
Comment (rleojoseph) [inline Sources/Classes/core/JavascriptBridge/MiniAppJavaScriptEnumerations.swift:6]: Thanks for adding here, `getContactId` name is certainly ambiguous

File: Sources/Classes/core/JavascriptBridge/MiniAppScriptMessageHandler.swift [hunk 3]
```diff
@@ -200,6 +205,36 @@ internal class MiniAppScriptMessageHandler: NSObject, WKScriptMessageHandler {
         }
     }
 
+    func getContactId(messageId: String) {
```
Comment (rleojoseph) [inline Sources/Classes/core/JavascriptBridge/MiniAppScriptMessageHandler.swift:208]: Can we rename it to `callbackId` ? Would be more precise

File: Sources/Classes/core/Protocols/MiniAppMessageDelegate.swift [hunk 4]
```diff
@@ -10,6 +10,12 @@ public protocol MiniAppMessageDelegate: MiniAppUserInfoDelegate, MiniAppShareCon
     /// Interface that should be implemented to return alphanumeric string that uniquely identifies a device.
     func getUniqueId(completionHandler: @escaping (Result<String?, MASDKError>) -> Void)
 
+    /// Interface that should be implemented to return alphanumeric string that uniquely identifies a device.
+    func getContactId(completionHandler: @escaping (Result<String?, MASDKError>) -> Void)
+
+    /// Interface that should be implemented to return alphanumeric string that uniquely identifies a device.
```
Comment (rleojoseph) [inline Sources/Classes/core/Protocols/MiniAppMessageDelegate.swift:16]: Same description 🤔 , the purchase is same. However do you think we can something extra here ?

**Review by rleojoseph — CHANGES_REQUESTED** [review]
Overall looks ok to me.
Could you please update the CHANGELOG, Userguide and fix Swiftlint issues (Making sure that we don't drop Codecov %)

File: Sources/Classes/core/JavascriptBridge/MiniAppJavaScriptEnumerations.swift (hunk 2 above)
Comment (laubengaier) [reply Sources/Classes/core/JavascriptBridge/MiniAppJavaScriptEnumerations.swift]: yeah I've replaced it according to the js sdk
uniqueId is still available but contactId is now messageUniqueId

File: Sources/Classes/core/JavascriptBridge/MiniAppScriptMessageHandler.swift (hunk 3 above)
Comment (laubengaier) [reply Sources/Classes/core/JavascriptBridge/MiniAppScriptMessageHandler.swift]: sure makes sense, updated it

File: Sources/Classes/core/Protocols/MiniAppMessageDelegate.swift (hunk 4 above)
Comment (laubengaier) [reply Sources/Classes/core/Protocols/MiniAppMessageDelegate.swift]: updated a slight difference do you think I should add more info about the assembly of the "id's"?

Comment (rleojoseph) [reply Sources/Classes/core/Protocols/MiniAppMessageDelegate.swift]: Looks ok now

**Review by corycaywood — APPROVED** [review]
Looks good 👍

**Review by rleojoseph — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**